tests/FactoryTest.php

```diff
@@ -25,6 +31,146 @@ protected function setUp(): void
         $this->container = new di\Container();
     }
 
+    protected function appConfig(): array
+    {
+        return [
+            'name' => 'some_name'
+        ];
+    }
+
+
+    public function testGetSession()
+    {
+        \Yii::$app = new \yii\web\Application([
+            'basePath' => __DIR__,
+            'id' => "some_id"
+        ]);
+
+        $this->getFunctionMock('yii\web', 'session_name')
+            ->expects($this->any())->willReturn("session_name1");
+
+        $this->assertEquals(
+            new \yii\web\Session([]),
+            $this->factory->getSession($this->container, [], [
```

Comment:
This assertion fill be true even when getSession() will return \yii\redis\Session